pydantic_ai_service.py

```
@@ -0,0 +1,107 @@
+            try:
+                # If the response is JSON, parse it directly
+                parsed_response = response_model.model_validate_json(content)
+                return parsed_response
+            except ValidationError as ve:
+                # If direct JSON parsing fails, try to extract structured data
+                # This handles cases where LLM returns text instead of pure JSON
+                try:
+                    # Attempt to create model from the text content
+                    parsed_response = response_model(contents=content)
+                    return parsed_response
+                except Exception:
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)